src/types.rs

```diff
@@ -248,6 +249,49 @@ impl From<JsonValue> for Value {
     }
 }
 
+impl From<Value> for JsonValue {
+    fn from(value: Value) -> Self {
+        match value {
+            Value::Null => JsonValue::Null,
+            Value::Boolean(b) => JsonValue::Bool(b),
+            Value::Int(i) => JsonValue::Number(i.into()),
+            Value::Long(l) => JsonValue::Number(l.into()),
+            Value::Float(f) => JsonValue::Number(Number::from_f64(f.into()).unwrap()),
+            Value::Double(d) => JsonValue::Number(Number::from_f64(d).unwrap()),
+            Value::Bytes(bytes) => JsonValue::Array(bytes.into_iter().map(|b| b.into()).collect()),
+            Value::String(s) => JsonValue::String(s),
+            Value::Fixed(_size, items) => JsonValue::Array(items.into_iter().map(|v| v.into()).collect()),
+            Value::Enum(_i, s) => JsonValue::String(s),
+            Value::Union(b) => (*b).into(),
+            Value::Array(items) => JsonValue::Array(items.into_iter().map(JsonValue::from).collect()),
+            Value::Map(items) => JsonValue::Object(
+                items
+                    .into_iter()
+                    .map(|(key, value)| (key, value.into()))
+                    .collect()
+            ),
+            Value::Record(items) => JsonValue::Object(
+                items
+                    .into_iter()
+                    .map(|(key, value)| (key, value.into()))
+                    .collect()
+            ),
+            Value::Date(d) => JsonValue::Number(d.into()),
+            Value::Decimal(d) => JsonValue::Array(d.to_vec().unwrap().into_iter().map(|v| v.into()).collect()),
+            //Value::Decimal(d) => JsonValue::Array(<Vec<u8>>::try_from(d).unwrap().into_iter().map(|v| v.into()).collect()),
```

Comment:
I'd like to switch to something like this to use the `try_from`, but currently I see " the trait `std::convert::From<decimal::Decimal>` is not implemented for `std::vec::Vec<u8>`"

Reply:
I'd rather avoid the use of `unwrap` in the library (I don't know if there is any `unwrap` left, but I should probably try to remove it if there is).
Feel free to implement it the trait you need, but you might probably want to implement `TryFrom` instead of `Try`.

Reply:
makes sense! I was following the pattern of From<JsonValue> above which uses `unwrap` instead of `TryFrom`: https://github.com/flavray/avro-rs/pull/155/files#diff-dfd26227c4842bd3288e9b1f53f0089fR236-R237. I'll switch to `TryFrom`